config/config/src/index.ts

```diff
@@ -639,3 +636,40 @@ function parsePackageManager (packageManager: string): { name: string, version:
     version,
   }
 }
+
+function addSettingsFromWorkspaceManifestToConfig (pnpmConfig: Config, {
+  configFromCliOpts,
+  projectManifest,
+  workspaceManifest,
+  workspaceDir,
+}: {
+  configFromCliOpts: Record<string, unknown>
+  projectManifest: ProjectManifest | undefined
+  workspaceDir: string | undefined
+  workspaceManifest: WorkspaceManifest
+}): void {
+  const newSettings = Object.assign(getOptionsFromPnpmSettings(workspaceDir, workspaceManifest, projectManifest), configFromCliOpts)
+  for (const [key, value] of Object.entries(newSettings)) {
+    if (!isCamelCase(key)) continue
+
+    // @ts-expect-error
+    pnpmConfig[key] = value
+
+    const kebabKey = kebabCase(key)
+    // Q: Why `types` instead of `rcOptionTypes`?
+    // A: `rcOptionTypes` includes options that would matter to the `npm` cli which wouldn't care about `pnpm-workspace.yaml`.
+    const targetKey = kebabKey in types ? kebabKey : key
+    pnpmConfig.rawConfig ??= {}
+    pnpmConfig.rawConfig[targetKey] = value
+  }
+  // All the pnpm_config_ env variables should override the settings from pnpm-workspace.yaml,
+  // as it happens with .npmrc.
+  // Until that is fixed, we should at the very least keep the right priority for verifyDepsBeforeRun,
+  // or else, we'll get infinite recursion.
+  // Related issue: https://github.com/pnpm/pnpm/issues/10060
+  if (process.env.pnpm_config_verify_deps_before_run != null) {
+    pnpmConfig.verifyDepsBeforeRun = process.env.pnpm_config_verify_deps_before_run as VerifyDepsBeforeRun
+    pnpmConfig.rawConfig['verify-deps-before-run'] = pnpmConfig.verifyDepsBeforeRun
+  }
+  pnpmConfig.catalogs = getCatalogsFromWorkspaceManifest(workspaceManifest)
```

Comment:
maybe we should have a list of settings that are allowed to be global. Like I am not sure it makes sense to have global catalogs.

Reply:
https://github.com/pnpm/pnpm/pull/10145/commits/acc2550055dcb535fd0c01c93f8fac234887118e